src/NUnitFramework/tests/Constraints/NumericsTest.cs

```diff
@@ -58,6 +58,58 @@ public void CanMatchDecimalWithoutToleranceMode()
             Assert.IsTrue(Numerics.AreEqual(123m, 123m, ref zeroTolerance));
         }
 
+        [Test]
+        public void CanCompareDecimalsWithHighPrecision()
+        {
+            var expected = 95217168582.206969750145956m;
+            var actual   = 95217168582.20696975014595521m;
+
+            var result = Numerics.Difference(expected, actual, ToleranceMode.Linear);
+
+            Assert.That(result, Is.EqualTo(0.00000000000000079M));
+        }
+
+        [Test]
+        public void TreatFloatingPointsSameForInstanceOrType()
+        {
+            var values = new object[] {
+                (double)1.1,
+                (float)1.1
+            };
+
+            Assert.Multiple(() =>
+            {
+                foreach (var value in values)
+                {
+                    Assert.That(Numerics.IsFloatingPointNumeric(value), Is.True);
+                    Assert.That(Numerics.IsFloatingPointNumeric(value.GetType()), Is.True);
+                }
+            });
+        }
+
+        [Test]
+        public void TreatFixedPointsSameForInstanceOrType()
+        {
+            var values = new object[] {
+                (byte)1,
+                (sbyte)1,
+                (decimal)1,
```

Comment:
These tests were more about documenting the existing implementation than any particular behaviour. Perhaps it's overkill to include them?